redis/pool_test.go

```diff
@@ -825,6 +825,30 @@ func TestWaitPoolGetContext(t *testing.T) {
 	defer c.Close()
 }
 
+func TestWaitPoolGetContextIssue520(t *testing.T) {
+	d := poolDialer{t: t}
+	p := &redis.Pool{
+		MaxIdle:   1,
+		MaxActive: 1,
+		Dial:      d.dial,
+		Wait:      true,
+	}
+	defer p.Close()
+	ctx1, _ := context.WithTimeout(context.Background(), 1*time.Nanosecond)
```

Comment:
This leaks a cancel see:
https://travis-ci.org/github/gomodule/redigo/builds/736087424